cartography/intel/gcp/compute.py

```diff
@@ -28,6 +28,124 @@
+            totalPages = int(totalPages)
+            if pageNo < totalPages or pageNo == totalPages:
+                logger.info(f'pages process for compute disks {pageNo}/{totalPages} pageSize is {pageSize}')
+            page_start = (common_job_parameters.get('pagination', {}).get('compute', None)[
+                          'pageNo'] - 1) * common_job_parameters.get('pagination', {}).get('compute', None)['pageSize']
+            page_end = page_start + common_job_parameters.get('pagination', {}).get('compute', None)['pageSize']
+            if page_end > len(disks) or page_end == len(disks):
+                disks = disks[page_start:]
+            else:
+                has_next_page = True
+                disks = disks[page_start:page_end]
+                common_job_parameters['pagination']['compute']['hasNextPage'] = has_next_page
+
+        return disks
+    except HttpError as e:
+        err = json.loads(e.content.decode('utf-8'))['error']
+        if err.get('status', '') == 'PERMISSION_DENIED' or err.get('message', '') == 'Forbidden':
+            logger.warning(
+                (
+                    "Could not retrieve compute disks on project %s due to permissions issues. Code: %s, Message: %s"
+                ), project_id, err['code'], err['message'],
+            )
+            return []
+        else:
+            raise
+
+
+@timeit
+def load_compute_disks(session: neo4j.Session, data_list: List[Dict], project_id: str, update_tag: int) -> None:
+    session.write_transaction(load_compute_disks_tx, data_list, project_id, update_tag)
+
+
+@timeit
+def load_compute_disks_tx(
+    tx: neo4j.Transaction, data: List[Dict],
+    project_id: str, gcp_update_tag: int,
+) -> None:
+
+    query = """
+    UNWIND {Records} as record
+    MERGE (disk:GCPComputeDisk{id:record.id})
+    ON CREATE SET
+        disk.firstseen = timestamp()
+    SET
+        disk.lastupdated = {gcp_update_tag},
+        disk.region = record.region,
+        disk.name = record.name,
+        disk.creation_timestamp = record.creationTimestamp,
+        disk.description = record.description,
+        disk.zone = record.zone,
+        disk.status = record.status,
+        disk.source_snapshot = record.sourceSnapshot,
+        disk.source_snapshot_id = record.sourceSnapshotId,
+        disk.source_storage_object = record.sourceStorageObject,
+        disk.options = record.options,
+        disk.self_link = record.selfLink,
+        disk.source_image = record.sourceImage,
+        disk.last_attach_timestamp = record.lastAttachTimestamp,
+        disk.last_detach_timestamp = record.lastDetachTimestamp,
+        disk.physical_block_size_bytes = record.physicalBlockSizeBytes,
+        disk.source_disk = record.sourceDisk,
+        disk.source_disk_id = record.sourceDiskId,
+        disk.source_image_id = record.sourceImageId
+    WITH disk
+    MATCH (owner:GCPProject{id:{ProjectId}})
+    MERGE (owner)-[r:RESOURCE]->(disk)
```

Comment:
We should build a relationship with GCPCompute Instances. Each instance can have multiple disks associated with it. So, let's represent that relationship between Compute Instance and Disk with `USES` relationship label.